java/ddlogapi.c

```diff
@@ -203,24 +291,30 @@ void commit_dump_callback(void* callbackInfo, table_id tableid, const ddlog_reco
         env, cbi->obj, cbi->method, (jint)tableid, (jlong)rec, (jboolean)polarity);
 }
 
-JNIEXPORT jint JNICALL Java_ddlogapi_DDlogAPI_ddlog_1transaction_1commit_1dump_1changes(
+JNIEXPORT void JNICALL Java_ddlogapi_DDlogAPI_ddlog_1transaction_1commit_1dump_1changes(
     JNIEnv * env, jobject obj, jlong handle, jstring callback) {
 
-    if (callback == NULL)
-        return ddlog_transaction_commit_dump_changes((ddlog_prog)handle, NULL, 0);
+    if (callback == NULL) {
+        if (ddlog_transaction_commit_dump_changes((ddlog_prog)handle, NULL, 0) < 0) {
+            throwDDlogException(env);
+            return;
+        }
+    }
 
     struct CallbackInfo* cbinfo = createCallback(env, obj, callback, "(IJZ)V");
     if (cbinfo == NULL)
-        return -1;
+        return;
```

Comment:
why not throw here?

Reply:
If `createCallback` failed, this means it has already thrown an exception.  We should just return and the exception will show up in Java.